File: ios/pfcz/Food.swift
import Foundation

struct Food: Identifiable, Codable {
    let id: UUID
    let name: String
    let protein: Double
    let fat: Double
    let carbs: Double
    let calories: Double
    
    init(id: UUID = UUID(), name: String, protein: Double, fat: Double, carbs: Double, calories: Double) {
        self.id = id
        self.name = name
        self.protein = protein
        self.fat = fat
        self.carbs = carbs
        self.calories = calories
    }
}

File: backend/app/models/food.rb
class Food < ApplicationRecord
  validates :name, presence: true
  validates :protein, presence: true, numericality: { greater_than_or_equal_to: 0 }
  validates :fat, presence: true, numericality: { greater_than_or_equal_to: 0 }
  validates :carbs, presence: true, numericality: { greater_than_or_equal_to: 0 }
  validates :calories, presence: true, numericality: { greater_than_or_equal_to: 0 }
end